hr_timesheet_holiday/models/hr_holidays.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+from openerp.exceptions import Warning as UserError
+
+
+class HrHolidaysStatus(models.Model):
+    """Add analytic account to holiday status"""
+    _inherit = 'hr.holidays.status'
+
+    analytic_account_id = fields.Many2one('account.analytic.account',
+        'Analytic Account')
+
+
+class HrHolidays(models.Model):
+    """Update analytic account on status change of Leave Request"""
+    _inherit = 'hr.holidays'
+
+    @api.multi
+    def holidays_validate(self):
+        res = super(HrHolidays, self).holidays_validate()
+
+        # Look for analytic account linked to Leave Type
+        for leave in self:
+            account = leave.holiday_status_id.analytic_account_id
+            if not account:
+                raise UserError("No analytic account defined for Leave Type '%s'" %
+                    (leave.holiday_status_id.name,))
+
+            # Look for company and hours per working day
+            if not leave.department_id:
+                raise UserError("Department not set")
+            company = leave.department_id.company_id
+            if not company:
```

Comment:
I did manage to unset it in the UI when multi-company mode was active, but I agree with you that it's very unlikely to encounter real world scenarios where employees will not have companies defined, so will remove it.